keras/utils/vis_utils.py

```
@@ -13,7 +13,13 @@
 
 
 def _check_pydot():
-    if not (pydot and pydot.find_graphviz()):
+    try:
+        if hasattr(pydot, 'find_graphviz'):
+            if not pydot.find_graphviz():
+                raise ImportError()
+        else:
+            pydot.Dot.create(pydot.Dot())
+    except:
```

No catch-all exceptions. You should catch a specific class.